pkg/infrastructure/mongodb.go

```diff
@@ -0,0 +1,279 @@
+	if IsMongoDBAvailable(cli) {
+		log.Debugf("MongoDB CRDs available. Checking if MongoDB Operator is deployed in the namespace %s", namespace)
+		// then check if there's an MongoDB Operator deployed
+		deployment := &v1.Deployment{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: MongoDBOperatorName}}
+		exists := false
+		var err error
+		if exists, err = kubernetes.ResourceC(cli).Fetch(deployment); err != nil {
+			return false, nil
+		}
+		if exists {
+			log.Debugf("MongoDB Operator is available in the namespace %s", namespace)
+			return true, nil
+		}
+	} else {
+		log.Debug("Couldn't find MongoDB CRDs")
+	}
+	log.Debugf("Looks like MongoDB Operator is not available in the namespace %s", namespace)
+	return false, nil
+}
+
+// GetMongoDBCredential gets the credential of the MongoDB server deployed with the Kogito Operator
+func GetMongoDBCredential(cli *client.Client, mongoDBInstance *mongodb.MongoDB) (*MongoDBCredential, error) {
+	user := findMongoDBUserByUsernameOrFirst(DefaultMongoDBUser, mongoDBInstance)
+	if user == nil {
+		return nil, fmt.Errorf("No user found for the given MongoDB Instance")
+	}
+
+	creds := &MongoDBCredential{
+		Username:     user.Name,
+		AuthDatabase: user.DB,
+	}
+
+	secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: user.PasswordSecretRef.Name, Namespace: mongoDBInstance.Namespace}}
+	if exists, err := kubernetes.ResourceC(cli).Fetch(secret); err != nil {
+		return nil, err
+	} else if exists {
+		passwordKey := defaultPasswordSecretRef
+		if user.PasswordSecretRef.Key != "" {
+			passwordKey = user.PasswordSecretRef.Key
+		}
+		creds.Password = string(secret.Data[passwordKey])
+	}
+
+	// Retrieve first role DB
+	creds.Database = DefaultMongoDBDatabase
+	if len(user.Roles) > 0 {
+		creds.Database = user.Roles[0].DB
+	}
+
+	return creds, nil
+}
+
+func findMongoDBUserByUsernameOrFirst(username string, mongoDBInstance *mongodb.MongoDB) *mongodb.MongoDBUser {
+	if len(mongoDBInstance.Spec.Users) <= 0 {
+		return nil
+	}
+	for _, user := range mongoDBInstance.Spec.Users {
+		if user.Name == username {
+			return &user
+		}
+	}
+	return &mongoDBInstance.Spec.Users[0]
+}
+
+// GetMongoDBDefaultSecret returns the default secret for MongoDB default user
+func GetMongoDBDefaultSecret(namespace string) *corev1.Secret {
+	return GetMongoDBSecret(namespace, DefaultMongoDBSecretName, DefaultMongoDBPassword)
+}
+
+// GetMongoDBSecret returns a MongoDB secret structure
+func GetMongoDBSecret(namespace, secretName, password string) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		StringData: map[string]string{
+			defaultPasswordSecretRef: password,
+		},
+	}
+}
+
+// GetMongoDBStub returns the preconfigured MongoDB stub with set namespace, name and secretName
+func GetMongoDBStub(openshift bool, namespace, name string, users []MongoDBUserCred) *mongodb.MongoDB {
```

Comment:
Dropping the requirement of creating the instance ourselves, I believe we will need to move this code to the BDD tests instead, since there I believe would make much more sense to have MongoDB installed automatically.